src/Sylius/Bundle/CoreBundle/Provider/SalesPerPeriodProvider.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Provider;
+
+use Doctrine\ORM\EntityRepository;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\OrderPaymentStates;
+use Sylius\Component\Core\Sales\Mapper\SalesPeriodMapperInterface;
+use Sylius\Component\Core\Sales\Provider\SalesPerPeriodProviderInterface;
+use Sylius\Component\Core\Sales\ValueObject\SalesPeriod;
+
+final class SalesPerPeriodProvider implements SalesPerPeriodProviderInterface
+{
+    /** @param EntityRepository<OrderInterface> $orderRepository */
+    public function __construct(private EntityRepository $orderRepository, private SalesPeriodMapperInterface $salesPeriodMapper)
+    {
+    }
+
+    public function provide(SalesPeriod $salesPeriod, ChannelInterface $channel): array
```

Comment:
In doctrine repositories filters, sortings or `PerPeriod` usually goes after base values.

```suggestion
    public function provide(ChannelInterface $channel, SalesPeriod $salesPeriod): array
```